### PR TITLE
Update core to v3.1.1

### PR DESCRIPTION
**Pull request by nrabinowitz**
* Updates H3 core to v3.1.1
* Fixes and hardens `h3SetToMultiPolygon` tests (to be less reliant on undefined algorithm characteristics)
* Drops `constrainLat` and `constrainLng`. These are no longer necessary for reads from the core lib, and never served great purpose as a guard against bad input from the caller.
* Updates CHANGELOG in prep for release

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,13 +3,17 @@
 All notable changes to this project will be documented in this file. This library adheres to a versioning policy described in [the README](./README.md#versioning). The public API of this library consists of the functions exported in [h3core.js](./lib/h3core.js).
 
 ## [Unreleased]
-- *None*
+### Fixed
+- Updated the core library to v3.1.1, including fixes for `polyfill` and `h3SetToMultiPolygon` (#20)
```
Comment (isaacbrodsky) [inline CHANGELOG.md]: #20 doesn't exist

Comment (nrabinowitz) [reply CHANGELOG.md]: Oh, good catch, will update to #19. Not sure why I thought this PR was going to be 20.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,11 +5,17 @@ All notable changes to this project will be documented in this file. This librar
 ## [Unreleased]
 - *None*
 
+### Fixed
+- Updated the core library to v3.1.1, including fixes for `polyfill` and `h3SetToMultiPolygon` (#19)
+- Removed Emscripten Node error handling from built library, fixing stacktraces (#18)
+### Added
+- Added generated API documentation to README (#19)
```
Comment (isaacbrodsky) [inline CHANGELOG.md]: This was #17

Comment (nrabinowitz) [reply CHANGELOG.md]: Ah, that must be why I was confused about #19. Sigh.

**Pull request merged**